### PR TITLE
Readd Changes for 25167 and Cleanup of CXF Upgrade

### PR DESCRIPTION
**Pull request by neuwerk**
This PR readds the mistakenly undone #25167 and address clean up related comments in #25447

### CODE REVIEW

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/bnd.bnd [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #*******************************************************************************
 -include=~../cnf/resources/bnd/bundle.props
 
-# This needs to stay at 3.2.4 even though the bundle version is 3.4.3.
+# This needs to stay at 3.2.4 even though the bundle version is 3.5.5.
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/bnd.bnd:15]: Did you try the feature checker ignore and it didn't work?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.hc.3.2/src/org/apache/cxf/transport/http/asyncclient/AsyncHTTPConduit.java [hunk 2]
```diff
@@ -573,7 +573,6 @@ protected void verifySession(final HttpHost host,
                     // See please https://issues.apache.org/jira/browse/HTTPASYNC-168, the attribute names
                     // are case-sensitive.
                     ctx.setAttribute("http.iosession-factory-registry", regBuilder.build());
-                    ctx.setAttribute("http.ioSession-factory-registry", regBuilder.build());
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.hc.3.2/src/org/apache/cxf/transport/http/asyncclient/AsyncHTTPConduit.java]: This is not a duplicate.  It says they are case sensitive and it is session vs Session.  I assumed this was supposed to stay?

**Review by jhanders34 — APPROVED** [review]

**Review by jhanders34 — COMMENTED** [review]
I removed approval based off of the comment I left where it seemed like it was supposed to leave the "duplicate" property name.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
Now selecting request changes so that github sees what I want it to see.

**Review by jhanders34 — APPROVED** [review]

**Comment by jhanders34** [comment]
#libby

**Comment by neuwerk** [comment]
#build

**Pull request merged**